iwyu_output.cc

```diff
@@ -605,6 +636,28 @@ void IwyuFileInfo::ReportForwardDeclareUse(SourceLocation use_loc,
   LogSymbolUse("Marked fwd-decl use of decl", symbol_uses_.back());
 }
 
+void IwyuFileInfo::ReportUsingDeclUse(SourceLocation use_loc,
+                                      const UsingDecl* using_decl,
+                                      const NamedDecl* target_decl,
+                                      bool in_cxx_method_body,
+                                      const char* comment) {  
+  for(auto&& saved_decl : using_decl_lines_) {
+    if(saved_decl.matches(using_decl)) {
+      saved_decl.set_referenced();
+      break;
+    }
+  }
+
+  ReportFullSymbolUse(use_loc, using_decl, in_cxx_method_body, comment);
```

Comment:
Cool, this looks much clearer!

A few things --

  * Isn't it a hard error If the `UsingDecl` is not found in the vector? I'd reorganize it so that `ReportFullSymbolUse` is called inside the loop, when the decl is found, and replace the `break` with a `return`. Then `CHECK_UNREACHABLE_` after the loop.
  * Can you add a comment above `ReportFullSymbolUse` to describe cases that make it necessary? I think it's for the so-called 'imported' usings from header files, is that it?